doc/test-framework/writing-e2e-tests.md

```diff
@@ -152,7 +163,7 @@ exampleMemcached := &cachev1alpha1.Memcached{
         Size: 3,
     },
 }
-err = f.DynamicClient.Create(goctx.TODO(), exampleMemcached)
+err = f.Client.Create(goctx.TODO(), exampleMemcached, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
```

Comment:
Can you also give an example snippet of:
```Go
err = f.Client.Create(goctx.TODO(), exampleMemcached, nil)
```
to highlight that this will Create and add a Cleanup function but not wait for the cleanup function to finish.

Reply:
Let's document all 3 cases for cleanup options clearly:

```Go
// Create with no cleanup
Create(goctx.TODO(), exampleMemcached, &framework.CleanupOptions{})
Create(goctx.TODO(), exampleMemcached, nil)

// Create with cleanup but no polling for resources to be deleted
Create(goctx.TODO(), exampleMemcached, &framework.CleanupOptions{TestContext: ctx})

// Create with cleanup and polling wait for resources to be deleted
Create(goctx.TODO(), exampleMemcached, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
```